src/test/java/com/loofah/graph/api/integrationTests/IntegrationTest.java

```diff
@@ -0,0 +1,71 @@
+package com.loofah.graph.api.integrationTests;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.loofah.graph.api.testConfig.GraphAPITestConfiguration;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = GraphAPITestConfiguration.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class IntegrationTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    private ObjectMapper objectMapper;
+
+    @Before
+    public void setUp() {
+        objectMapper = new ObjectMapper();
+    }
+
+    @Test
+    public void returns_all_skills() throws IOException {
+
+        String query = "{allSkills{id, title, description}}";
+        String uri = "http://localhost:" + port + "/skills?query={query}";
+        ResponseEntity<String> response = testRestTemplate.getForEntity(uri, String.class, query);
+
+        JsonNode parsedResponseBody = objectMapper.readTree(response.getBody()).get("data").get("allSkills");
+        List<LinkedHashMap> allSkills = objectMapper.convertValue(parsedResponseBody, ArrayList.class);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(3, allSkills.size());
```

Comment:
This test feels pretty flakey, but age-old debate about how much we want to API test from our integration tests?

Reply:
I'm personally happy with this level of integration testing, especially if we have good unit tests and aim to set up some end-to-end tests